src/components/MovieDetails/index.js

```diff
@@ -10,8 +10,10 @@ class MovieDetails extends React.Component {
         )
     }
     render() {
-        const { movie } = this.props;
-        const { title, poster_path, release_date, overview, tagline, runtime } = movie;
+
+        const { match, movies } = this.props;
+        const id = parseInt(match.params.id, 10)
+        const { title, poster_path, release_date, overview, tagline, runtime } = movies.filter(item => item.id == id)[0];
```

Comment:
why not use movies.find instead of filter(...)[0]? and also if no match found destructing of undefined value will fail